ord_schema/validations.py

```diff
@@ -261,10 +286,39 @@ def validate_dataset(message):
         for reaction_id in message.reaction_ids:
             if not re.fullmatch('^ord-[0-9a-f]{32}$', reaction_id):
                 warnings.warn('Reaction ID is malformed', ValidationError)
-    if message.dataset_id:
+    if validate_id:
         # The dataset_id is a 32-character uuid4 hex string.
         if not re.fullmatch('^ord_dataset-[0-9a-f]{32}$', message.dataset_id):
             warnings.warn('Dataset ID is malformed', ValidationError)
+    # Check cross-references
```

Comment:
Let me know if there is a more concise way to do these checks. I realize this is an absurd level of nesting, but I find it legible at least.

Reply:
This is good; maybe factor out as a separate method just to keep `validate_dataset` sane?

Reply:
I made a separate `get_referenced_reaction_ids` that makes this a little cleaner